src/test/kotlin/vonage/spring/VonageArgumentResolverTest.kt

```diff
@@ -0,0 +1,117 @@
+package vonage.spring
+
+import com.vonage.client.incoming.CallDirection
+import com.vonage.client.incoming.CallEvent
+import com.vonage.client.incoming.CallStatus
+import com.vonage.client.incoming.MessageEvent
+import org.junit.Test
+import org.springframework.core.MethodParameter
+import org.springframework.mock.web.MockHttpServletRequest
+import org.springframework.web.context.request.ServletWebRequest
+import java.text.SimpleDateFormat
+import java.time.Instant
+import java.util.*
+import kotlin.test.assertEquals
+
+class VonageArgumentResolverTest {
+
+    fun ExampleFunction(@VonageWebhook msgEvent: MessageEvent){
+
+    }
+
+    fun ExampleFunction2(@VonageWebhook callEvent: CallEvent){
+
+    }
+
+    @Test
+    fun testArgumentResolverWithUrlContent(){
+        val request = MockHttpServletRequest()
+        request.addParameter("msisdn","12018675309")
+        request.addParameter("to","19738675309")
+        request.addParameter("text", "Testing get type")
+        request.addParameter("messageId","17000002A9E9F4E0")
+        request.addParameter("keyword","TESTING")
+        request.addParameter("message-timestamp","2021-03-18 19:04:59")
+        request.addParameter("timestamp","1616094479")
+        request.addParameter("nonce","231a9c15-6732-4ea7-81d1-3e564f3a89b2")
+        val method = VonageArgumentResolverTest::class.java.declaredMethods.find { m->m.name == "ExampleFunction" }
+        val methodParam = MethodParameter(method,0)
+        val resolver = VonageArgumentResolver()
+        val nwr = ServletWebRequest(request)
+        val msg = resolver.resolveArgument(methodParam, null,nwr,null) as MessageEvent
+        assertEquals("12018675309", msg.msisdn)
+        assertEquals("19738675309", msg.to)
+        assertEquals("Testing get type", msg.text)
+        assertEquals("17000002A9E9F4E0", msg.messageId)
+        assertEquals("TESTING",msg.keyword)
+        assertEquals(SimpleDateFormat("yyyy-MM-dd HH:mm:ss").parse("2021-03-18 19:04:59"), msg.messageTimestamp)
+        assertEquals("1616094479", msg.timestamp)
+        assertEquals("231a9c15-6732-4ea7-81d1-3e564f3a89b2", msg.nonce)
+    }
+
+    @Test
+    fun testArgumentResolverWith(){
+        val request = MockHttpServletRequest()
+        request.contentType="application/json"
+        val json: String = "{\n" +
```

Comment:
Take a look at kotlin triple quited strings `"""abc"""`
https://kotlinlang.org/docs/basic-types.html#string-literals

Reply:
this I did actually steal directly from our Java Code 😆